apis/kueue/v1alpha2/clusterqueue_types.go

```diff
@@ -124,6 +124,9 @@ type ClusterQueueSpec struct {
 
 	// QueueingStrategy indicates the queueing strategy of the workloads
 	// across the queues in this ClusterQueue. This field is immutable.
+	// Since Kubernetes 1.25, we can use CEL validation rules to implement
+	// a few common immutability patterns directly in the manifest for a CRD.
+	// ref: https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/
```

Comment:
This is part of the user documentation, don't add this kind of detail here. Maybe leave it in the webhook code.